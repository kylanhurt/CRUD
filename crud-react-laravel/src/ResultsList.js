import React, { Component } from 'react'
import { Table } from 'semantic-ui-react'
import _ from 'lodash'

const API_URL = 'http://127.0.0.1:8000/api'

class ResultsList extends Component {
    constructor(props) {
        super(props);
        this.state = {
            peopleData: [],
            groupData: [],
            peopleUploadFeedback: {
                message: '',
                type: ''
            },
            groupsUploadFeedback: {
                message: '',
                type: ''
            },
            groupColumn: null,
            groupDirection: null,
            peopleColumn: null,
            peopleDirection: null
        };
    }

    componentDidMount  = async () => {
        const peopleResponse = await fetch("http://localhost:8000/api/people")
        const peopleData = await peopleResponse.json()
        this.setState({
            peopleData: peopleData.data
        })
    }

    onPeopleFileSelected = async (event) => {
        // console.log(event.target.files[0])
        const formData = new FormData()
        formData.append('file', event.target.files[0])
        const uploadPeopleResponse = await fetch(API_URL + '/people', {
            method: 'POST',
            body: formData
        })
        if (uploadPeopleResponse.ok) {
            const responseData = await uploadPeopleResponse.json()
            this.setState({
                peopleData: responseData.collection.data,
                peopleUploadFeedback: {
                    message: responseData.feedback,
                    type: 'success'
                }
            })
        } else {
            this.setState({
                peopleUploadFeedback: 'Some error'
            })
        }
    }

    onGroupFileSelected = async (event) => {
        // console.log(event.target.files[0])
        const formData = new FormData()
        formData.append('file', event.target.files[0])
        const uploadGroupResponse = await fetch(API_URL + '/groups', {
            method: 'POST',
            body: formData
        })
        if (uploadGroupResponse.ok) {
            const responseData = await uploadGroupResponse.json()
            this.setState({
                groupData: responseData.collection.data,
                groupsUploadFeedback: {
                    message: responseData.feedback,
                    type: 'success'
                }
            })
        } else {
            this.setState({
                groupsUploadFeedback: 'Some error'
            })
        }
    }

    handlePeopleSort = (clickedColumn) => () => {
        const { peopleColumn, peopleData, peopleDirection } = this.state

        if (peopleColumn !== clickedColumn) {
          this.setState({
            peopleColumn: clickedColumn,
            peopleData: _.sortBy(peopleData, [clickedColumn]),
            peopleDirection: 'ascending',
          })

          return
        }

        this.setState({
          peopleData: peopleData.reverse(),
          peopleDirection: peopleDirection === 'ascending' ? 'descending' : 'ascending',
        })
    }

    handleGroupSort = (clickedColumn) => () => {
        const { groupColumn, groupData, groupDirection } = this.state

        if (groupColumn !== clickedColumn) {
          this.setState({
            groupColumn: clickedColumn,
            groupData: _.sortBy(groupData, [clickedColumn]),
            groupDirection: 'ascending',
          })

          return
        }

        this.setState({
          groupData: groupData.reverse(),
          groupDirection: groupDirection === 'ascending' ? 'descending' : 'ascending',
        })
    }

    render() {
        const {
            peopleUploadFeedback,
            groupsUploadFeedback,
            peopleData,
            groupData,
            peopleDirection,
            peopleColumn,
            groupDirection,
            groupColumn
        } = this.state
        return (
            <div>
                <div>
                    <label htmlFor="peopleCSVUpload">Upload a people list:</label><br /><br />
                    <input type="file"
                        id="peopleCSVUpload" name="peopleCSVUpload"
                        accept="text/csv"
                        onChange={this.onPeopleFileSelected}
                    />
                </div>
                <br />
                <div style={{ minHeight: 40 }}>
                    <span style={{ color: peopleUploadFeedback.type === 'success' ? 'green' : 'red' }}>
                        {peopleUploadFeedback.message}
                    </span>
                </div>
                <div>
                    <label htmlFor="groupCSVUpload">Upload a group list:</label><br /><br />
                    <input type="file"
                        id="groupCSVUpload" name="groupCSVUpload"
                        accept="text/csv"
                        onChange={this.onGroupFileSelected}
                    />
                </div>
                <br />
                <div style={{ minHeight: 40 }}>
                    <span style={{ color: groupsUploadFeedback.type === 'success' ? 'green' : 'red' }}>
                        {groupsUploadFeedback.message}
                    </span>
                </div>
                    <Table id="peopleTable" celled padded sortable style={{ marginBottom: 60 }}>
                        <Table.Header>
                            <Table.Row>
                            <Table.HeaderCell singleLine sorted={peopleColumn === 'first_name' ? peopleDirection : null} onClick={this.handlePeopleSort('first_name')}>First Name</Table.HeaderCell>
                            <Table.HeaderCell sorted={peopleColumn === 'last_name' ? peopleDirection : null} onClick={this.handlePeopleSort('last_name')}>Last Name</Table.HeaderCell>
                            <Table.HeaderCell sorted={peopleColumn === 'email_address' ? peopleDirection : null} onClick={this.handlePeopleSort('email_address')}>Email</Table.HeaderCell>
                            <Table.HeaderCell sorted={peopleColumn === 'status' ? peopleDirection : null} onClick={this.handlePeopleSort('status')}>Status</Table.HeaderCell>
                            </Table.Row>
                        </Table.Header>

                        <Table.Body>

                        {
                            peopleData.map((person, index) => {
                                return (
                                    <Table.Row key={index}>
                                        <Table.Cell singleLine>{ person.first_name }</Table.Cell>
                                        <Table.Cell singleLine>{ person.last_name }</Table.Cell>
                                        <Table.Cell singleLine>{ person.email_address }</Table.Cell>
                                        <Table.Cell singleLine>{ person.status }</Table.Cell>
                                    </Table.Row>
                                );
                            })
                        }

                        </Table.Body>
                    </Table>
                    <Table id="groupsTable" celled padded sortable>
                    <Table.Header>
                        <Table.Row>
                            <Table.HeaderCell sorted={groupColumn === 'id' ? groupDirection : null} onClick={this.handleGroupSort('id')}>Group ID</Table.HeaderCell>
                            <Table.HeaderCell singleLine sorted={groupColumn === 'group_name' ? groupDirection : null} onClick={this.handleGroupSort('group_name')}>Group Name</Table.HeaderCell>
                        </Table.Row>
                    </Table.Header>

                    <Table.Body>

                    {
                        groupData.map((group, index) => {
                            return (
                                <Table.Row key={index}>
                                    <Table.Cell singleLine>{ group.id }</Table.Cell>
                                    <Table.Cell singleLine>{ group.group_name }</Table.Cell>
                                </Table.Row>
                            );
                        })
                    }

                    </Table.Body>
                </Table>
            </div>
    );
}

}

export default ResultsList
