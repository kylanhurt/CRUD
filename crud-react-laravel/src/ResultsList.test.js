import React from 'react';
import { shallow, configure } from 'enzyme';
import ResultsList from './ResultsList';
import Adapter from 'enzyme-adapter-react-16'
import "core-js/stable"
import "regenerator-runtime/runtime"

let wrapper, data;
configure({adapter: new Adapter()});

describe('<ResultsList />', () => {

    beforeAll(() => {
        wrapper = shallow(<ResultsList />)
        data = [{
            "id": 132,
            "first_name": "Macie",
            "last_name": "Emmerich",
            "email_address": "cremin.marjory@hotmail.com",
            "status": "active",
            "updated_at": "2019-07-20 22:05:47",
            "created_at": "2019-07-20 22:05:47"
        }]

        wrapper.setState({ 'peopleData' : data });
    });

    test('should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
})
