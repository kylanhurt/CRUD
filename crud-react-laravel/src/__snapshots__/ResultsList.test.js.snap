// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultsList /> should match the snapshot 1`] = `
<div>
  <div>
    <label
      htmlFor="peopleCSVUpload"
    >
      Upload a people list:
    </label>
    <br />
    <br />
    <input
      accept="text/csv"
      id="peopleCSVUpload"
      name="peopleCSVUpload"
      onChange={[Function]}
      type="file"
    />
  </div>
  <br />
  <div
    style={
      Object {
        "minHeight": 40,
      }
    }
  >
    <span
      style={
        Object {
          "color": "red",
        }
      }
    />
  </div>
  <div>
    <label
      htmlFor="groupCSVUpload"
    >
      Upload a group list:
    </label>
    <br />
    <br />
    <input
      accept="text/csv"
      id="groupCSVUpload"
      name="groupCSVUpload"
      onChange={[Function]}
      type="file"
    />
  </div>
  <br />
  <div
    style={
      Object {
        "minHeight": 40,
      }
    }
  >
    <span
      style={
        Object {
          "color": "red",
        }
      }
    />
  </div>
  <Table
    as="table"
    celled={true}
    id="peopleTable"
    padded={true}
    sortable={true}
  >
    <TableHeader
      as="thead"
    >
      <TableRow
        as="tr"
        cellAs="td"
      >
        <TableHeaderCell
          as="th"
          onClick={[Function]}
          singleLine={true}
          sorted={null}
        >
          First Name
        </TableHeaderCell>
        <TableHeaderCell
          as="th"
          onClick={[Function]}
          sorted={null}
        >
          Last Name
        </TableHeaderCell>
        <TableHeaderCell
          as="th"
          onClick={[Function]}
          sorted={null}
        >
          Email
        </TableHeaderCell>
        <TableHeaderCell
          as="th"
          onClick={[Function]}
          sorted={null}
        >
          Status
        </TableHeaderCell>
      </TableRow>
    </TableHeader>
    <TableBody
      as="tbody"
    >
      <TableRow
        as="tr"
        cellAs="td"
        key="0"
      >
        <TableCell
          as="td"
          singleLine={true}
        >
          Macie
        </TableCell>
        <TableCell
          as="td"
          singleLine={true}
        >
          Emmerich
        </TableCell>
        <TableCell
          as="td"
          singleLine={true}
        >
          cremin.marjory@hotmail.com
        </TableCell>
        <TableCell
          as="td"
          singleLine={true}
        >
          active
        </TableCell>
      </TableRow>
    </TableBody>
  </Table>
</div>
`;
